File: BEENUMS/beenums.c
#include <stdio.h>

int main() {
	int n;
	while(scanf("%d", &n) == 1 && n!= -1) {
		int u = 1;
		if(n==1)
			printf("Y\n");
		else {
			int i=0;
			while(u<n) {
				u+=6*i;
				i++;
			}
			if(u == n) {
				printf("Y\n");
			}
			else printf("N\n");
		}
	}
	return 0;
}

File: BEENUMS/beenums.go
package main

import "fmt"

func main() {
	var a int
	for ; a != -1; fmt.Scanf("%d", &a) {
		if a == 1 {
			fmt.Println("Y")
		} else {
			i := 0
			u := 1
			for u < a {
				u += 6 * i
				i++
			}
			if u == a {
				fmt.Println("Y")
			} else {
				fmt.Println("N")
			}
		}
	}
}
